src/main.ts

```diff
@@ -372,43 +372,54 @@ export default class AdvancedURI extends Plugin {
     }
 
     async handleWrite(parameters: Parameters, createdDailyNote: boolean = false) {
-        const path = parameters.filepath;
-        const file = this.app.vault.getAbstractFileByPath(path);
-        let outFile: TFile;
-        if (parameters.mode === "overwrite") {
-            outFile = await this.writeAndOpenFile(path, parameters.data, parameters);
-            this.success(parameters);
-        } else if (parameters.mode === "prepend") {
-            if (file instanceof TFile) {
-                outFile = await this.prepend(file, parameters);
-            } else {
-                outFile = await this.prepend(path, parameters);
-            }
-            this.success(parameters);
-        } else if (parameters.mode === "append") {
-            if (file instanceof TFile) {
-                outFile = await this.append(file, parameters);
-            } else {
-                outFile = await this.append(path, parameters);
-            }
-            this.success(parameters);
-        } else if (parameters.mode === "new") {
-            if (file instanceof TFile) {
-                outFile = await this.writeAndOpenFile(this.getAlternativeFilePath(file), parameters.data, parameters);
-                this.hookSuccess(parameters, outFile);
+        let file: TAbstractFile;
+        if (parameters.filepath) {
+            file = this.app.vault.getAbstractFileByPath(parameters.filepath);
+        } else {
+            file = this.app.workspace.getActiveFile();
+        }
+
+        if (file) {
```

Comment:
The method should write to non existent files too. That's the reason for using `path` instead of `file` in some places in the method.

Reply:
Oops, forgot that `getAbstractFileByPath` returns `null` sometimes when the file cannot be found. `parameters.filepath` set now always allows the writing logic to be triggered.